yaptide/simulation_runner/shieldhit_runner.py

```diff
@@ -10,108 +10,91 @@
 
-        for config_filename in input_dict:
-            abs_input_path = os.path.join(tmp_output_path, config_filename)
-            with open(abs_input_path, 'w') as temp_input_file:
-                temp_input_file.write(input_dict[config_filename])
+        # digest dictionary with project data (extracted from JSON file)
+        # and generate SHIELD-HIT12A input files
+        conv_runner = ConvRunner(parser=DummmyParser(),
+                                 input_data=raw_input_dict,
+                                 output_dir=tmp_output_path)
+
+        conv_runner.run_parser()
 
         settings = SimulationSettings(input_path=tmp_output_path,
                                       simulator_exec_path=None,
                                       cmdline_opts='')
 
-        runner_obj = Runner(jobs=param_dict['jobs'],
-                            keep_workspace_after_run=False,
-                            output_directory=tmp_output_path)
+        runner_obj = SHRunner(jobs=param_dict['jobs'],
+                              keep_workspace_after_run=False,
+                              output_directory=tmp_output_path)
 
         start_time = timeit.default_timer()
+
         isRunOk = runner_obj.run(settings=settings)
         if not isRunOk:
             return None
 
         elapsed = timeit.default_timer() - start_time
         print("MC simulation took {:.3f} seconds".format(elapsed))
 
-        estimator = runner_obj.get_data()
-
-        return estimator
-
-
-if __name__ == '__main__':
-    sys.exit(run_shieldhit(sys.argv[1:]))
+        estimators_dict = runner_obj.get_data()
+
+        return dummy_convert_output(estimators_dict)
+
+
+def dummy_convert_output(estimators_dict):
+    """Dummy function for converting simulation output to dictionary"""
+    if not estimators_dict:
+        return {"message": "No estimators"}
+
+    # result_dict is the dictionary object, which is later converted to json
+    # to provide readable api response for fronted
+
+    # result_dict contains the list of estimators
+    result_dict = {"estimators": []}
+    for estimator in estimators_dict:
+
+        # estimator_dict contains list of pages
+        estimator_dict = {
+            "name" : estimator,
+            "pages": []}
+        for page in estimators_dict[estimator].pages:
+
+            page_dim = page.dimension
+
+            # handling 1 dimension page
+            if page_dim == 1:
+                axis = page.plot_axis(0)
+                x_values = axis.data
+                y_values = page.data_raw.flatten()
+
+                # for 1 dimension page, dict contains:
+                # "dimensions" indicating it is 1 dim page
+                # "unit"
+                # "x_y_pairs" which is list of x any y values pairs
+                page_dict = {
+                    "dimensions" : page_dim,
+                    "unit": str(axis.unit),
+                    "x_y_pairs": []
```

Comment:
Why are you creating list of (x,y) pairs ?
Looking at:
![image](https://user-images.githubusercontent.com/7374733/139734048-db1098ea-3be4-401c-b7f9-9f6de79e674c.png)
https://github.com/yaptide/ui/pull/226/files#diff-58cddae6837f8e66ec955968adbef3da86fb51763ec55e990dce8398b3d0131cR23
it seems that frontend prefers list of x and list of y values, separately.

Reply:
Changed